### PR TITLE
Refactor and add more materials

### PR DESCRIPTION
**Pull request by markbandstra**
This PR will add additional materials commonly used in radiation modeling from the following reference:

~~McConn, Gesh, Pagh, Rucker, & Williams, "Compendium of Material Composition Data for Radiation Transport Modeling", PIET-43741-TM-963, PNNL-15870 Rev. 1~~

~~which is available [here](https://www.pnnl.gov/main/publications/external/technical_reports/PNNL-15870Rev1.pdf).~~
    
    Detwiler, Rebecca S., McConn, Ronald J., Grimes, Thomas F., Upton, Scott A., & Engel, Eric J., "Compendium of Material Composition Data for Radiation Transport Modeling". United States. PNNL-15870 Revision 2.

which is available in PDF and JSON form [here](https://compendium.cwmd.pnnl.gov).

~~The additional materials from this document will require an optional dependency on [PyPDF2](https://github.com/mstamy2/PyPDF2) to parse the data from the PDF document.~~

Additional features in this PR:
* Materials data will be cached locally in a simple CSV format, so it only needs to be retrieved once. Currently, internet access is required every time one wishes to access the NIST material data.
* Materials data will be refactored so that it is available as a `dict` of `dict`s, making access to each individual material's data easier than the current `pandas.DataFrame` approach.

Items to complete:
- [x] Raise warning if Compendium data cannot be parsed, provide details on how to access it
- ~~[ ] Update README to explain optional dependency~~
- [x] Make test coverage more complete
- [x] Provide option for clearing materials cache
- [x] Update to new 2021 version of the Compendium
- [x] Remove debugging outputs
- [x] Update notebooks

### CODE REVIEW

**Comment by markbandstra** [comment]
@jvavrek helpfully noted that there is a [new revision of the Compendium available](https://www.osti.gov/biblio/1782721), and it is even available on a [website](https://compendium.cwmd.pnnl.gov) for download as JSON. This will make this PR a bit easier and eliminate the need for parsing the PDF.

File: becquerel/tools/materials.py [hunk 1]
```diff
@@ -1,202 +1,253 @@
...
+        ):
+            print(tokens)
+            if tokens[0].startswith("%"):
+                continue
+            name = tokens[0]
+            formula = tokens[1]
+            density = float(tokens[2])
+            z_over_a = float(tokens[3])
+            weight_fracs = tokens[4].split(";")
+            source = ",".join(tokens[5:])
```
Comment (markbandstra) [inline becquerel/tools/materials.py:221]: This looks unnecessary; `csv.reader` should respect double quotes.

File: becquerel/tools/materials.py [hunk 2]
```diff
@@ -1,202 +1,218 @@
...
+            for k in range(len(weight_fracs1)):
+                elem1, frac1 = weight_fracs1[k].split(" ")
+                elem2, frac2 = weight_fracs2[k].split(" ")
+                if elem1 != elem2:
+                    raise MaterialsError(
+                        f"Material {name} weight fraction elements do not match between different data sources:  {elem1}  {elem2}"
+                    )
+                frac1 = float(frac1)
+                frac2 = float(frac2)
+            if not np.isclose(frac1, frac2, atol=3e-4):
```
Comment (jvavrek) [inline becquerel/tools/materials.py:63]: Is there a reason the `atol` are different for different properties?

Comment (markbandstra) [reply becquerel/tools/materials.py:63]: Yes, some of the data have changed in the Compendium versus the same materials in the NIST datasets. Those tolerances are just to show that the numbers are generally close but were found by increasing until the tests passed.

**Review by jvavrek — APPROVED** [review]
Nice, just a few comments

File: becquerel/tools/materials.py [hunk 3]
```diff
@@ -1,202 +1,218 @@
...
-    Raises:
-      NISTMaterialsRequestError: if there was a problem obtaining the data.
-
+    # fetch the data sources
+    data_elem = fetch_element_data()
+    data_mat = fetch_compound_data()
+    data_comp = fetch_compendium_data()
+
+    # perform various checks on the Compendium data
+    for j in range(len(data_comp)):
```
[the remark this answers is not in the document]
Comment (markbandstra) [reply becquerel/tools/materials.py:29]: Thanks, good point.

File: becquerel/tools/materials.py [hunk 4]
```diff
@@ -1,202 +1,218 @@
...
+    """Fetch all available materials.
 
-    Returns:
-      A list of strings of composition by symbol, e.g.,
-        ["H 0.111898", "O 0.888102"]
+    On first ever function call, will check NIST website for data using
+    the tools in materials_nist.py and will download and attempt to parse
+    the PNNL Compendium using the tools in materials_compendium.py.
+    The Compendium materials will only be available if the optional dependency
+    PyPDF2 package is installed.
```
[the remark this answers is not in the document]
Comment (markbandstra) [reply becquerel/tools/materials.py:197]: Nope, I'll remove it!

File: becquerel/tools/materials_compendium.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            MaterialsWarning,
+        )
+        data = []
+    else:
+        with open(FNAME, "r") as f:
+            data = json.load(f)
+
+    # extract relevant data
+    names = [datum["Name"] for datum in data]
+    formulae = [datum["Formula"] if "Formula" in datum else "-" for datum in data]
```
[the remark this answers is not in the document]
Comment (markbandstra) [reply becquerel/tools/materials_compendium.py:68]: Always

File: becquerel/tools/materials_nist.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+
+def fetch_element_data():
+    """Retrieve data for the elements.
+
+    Data are found in Table 1:
+      http://physics.nist.gov/PhysRefData/XrayMassCoef/tab1.html
+
+    Returns:
+      A pandas DataFrame of the material data. Colums are 'Z', 'Symbol',
```
[the remark this answers is not in the document]
Comment (markbandstra) [reply becquerel/tools/materials_nist.py:55]: Oof, that's an old error.

**Pull request merged**